PowerPointLabs/PowerPointLabs/SyncLab/SyncLabShapeStorage.cs

```diff
@@ -9,13 +9,25 @@ namespace PowerPointLabs.SyncLab
     public class SyncLabShapeStorage : PowerPointPresentation
     {
 
+        public const int FormatStorageSlide = 0;
+
         private int nextKey = 0;
 
-        public SyncLabShapeStorage() : base()
+        private static SyncLabShapeStorage storage;
+
+        public static SyncLabShapeStorage GetInstance()
+        {
+            if (storage == null)
```

Comment:
it's rare the there can be race condition here, do u want to lock/sync this method?